`EACCES: permission denied, mkdir '/home/runner' occurred on a self-hosted runner

On a self-hosted runner, the following error occurred.

```
Run engineerd/setup-kind@v0.3.0
downloading kind from https://github.com/kubernetes-sigs/kind/releases/download/v0.7.0/kind-linux-amd64
##[error]EACCES: permission denied, mkdir '/home/runner'
```

--- a/.github/workflows/e2e.yml
+++ b/.github/workflows/e2e.yml
@@ -6,7 +6,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@master
-      - uses: engineerd/setup-kind@v0.3.0
+      - name: "npm install"
+        run: npm install
+      - name: "npm run build"
+        run: npm run build
+      - uses: ./
       - name: Testing
         run: |
           kubectl cluster-info

--- a/src/kind.ts
+++ b/src/kind.ts
@@ -70,13 +70,19 @@ export function getKindConfig(): KindConfig {
 
 // this action should always be run from a Linux worker
 export async function downloadKind(version: string) {
-    let url: string = `https://github.com/kubernetes-sigs/kind/releases/download/${version}/kind-linux-amd64`;
-    console.log("downloading kind from " + url);
-    let downloadPath: string | null = null;
-    downloadPath = await tc.downloadTool(url);
-    await exec.exec("chmod", ["+x", downloadPath]);
-    let toolPath = await tc.cacheFile(downloadPath, "kind", "kind", version);
-    core.debug(`kind is cached under ${toolPath}`);
+    let toolPath: string = tc.find("kind", version);
+
+    if (toolPath !== "") {
+        console.log(`found a cached file in ${toolPath}`)
+    } else {
+        let url: string = `https://github.com/kubernetes-sigs/kind/releases/download/${version}/kind-linux-amd64`;
+        console.log("downloading kind from " + url);
+        let downloadPath: string | null = null;
+        downloadPath = await tc.downloadTool(url);
+        await exec.exec("chmod", ["+x", downloadPath]);
+        toolPath = await tc.cacheFile(downloadPath, "kind", "kind", version);
+        core.debug(`kind is cached under ${toolPath}`);
+    }
 
     core.addPath(toolPath);
 }

--- a/src/kind.ts
+++ b/src/kind.ts
@@ -11,6 +11,8 @@ const NameInput: string = "name";
 const WaitInput: string = "wait";
 const SkipClusterCreationInput: string = "skipClusterCreation";
 
+const toolName: string = "kind";
+
 export class KindConfig {
     version: string
     configFile: string;
@@ -69,21 +71,24 @@ export function getKindConfig(): KindConfig {
 }
 
 // this action should always be run from a Linux worker
-export async function downloadKind(version: string) {
-    let toolPath: string = tc.find("kind", version);
+export async function downloadKind(version: string): Promise<string> {
+    let url: string = `https://github.com/kubernetes-sigs/kind/releases/download/${version}/kind-linux-amd64`;
+    console.log("downloading kind from " + url);
+    let downloadPath: string | null = null;
+    downloadPath = await tc.downloadTool(url);
+    await exec.exec("chmod", ["+x", downloadPath]);
+    let toolPath: string = await tc.cacheFile(downloadPath, "kind", toolName, version);
+    core.debug(`kind is cached under ${toolPath}`);
 
-    if (toolPath !== "") {
-        console.log(`found a cached file in ${toolPath}`)
-    } else {
-        let url: string = `https://github.com/kubernetes-sigs/kind/releases/download/${version}/kind-linux-amd64`;
-        console.log("downloading kind from " + url);
-        let downloadPath: string | null = null;
-        downloadPath = await tc.downloadTool(url);
-        await exec.exec("chmod", ["+x", downloadPath]);
-        toolPath = await tc.cacheFile(downloadPath, "kind", "kind", version);
-        core.debug(`kind is cached under ${toolPath}`);
-    }
-
-    core.addPath(toolPath);
+    return toolPath;
 }
 
+export async function getKind(version: string): Promise<string> {
+  let toolPath: string = tc.find(toolName, version);
+
+  if (toolPath === "") {
+    toolPath = await downloadKind(version);
+  }
+
+  return toolPath;
+}

--- a/src/kind.ts
+++ b/src/kind.ts
@@ -74,11 +74,10 @@ export async function downloadKind(version: string) {
     console.log("downloading kind from " + url);
     let downloadPath: string | null = null;
     downloadPath = await tc.downloadTool(url);
-    const binPath: string = "/home/runner/bin";
-    await io.mkdirP(binPath);
     await exec.exec("chmod", ["+x", downloadPath]);
-    await io.mv(downloadPath, path.join(binPath, "kind"));
+    let toolPath = await tc.cacheFile(downloadPath, "kind", "kind", version);
+    core.debug(`kind is cached under ${toolPath}`);
 
-    core.addPath(binPath);
+    core.addPath(toolPath);
 }
 

--- a/src/main.ts
+++ b/src/main.ts
@@ -1,10 +1,11 @@
 import * as core from '@actions/core';
-import { KindConfig, getKindConfig, downloadKind } from './kind';
+import { KindConfig, getKindConfig, getKind } from './kind';
 
 async function run() {
   try {
     let cfg: KindConfig = getKindConfig();
-    await downloadKind(cfg.version);
+    let toolPath: string = await getKind(cfg.version);
+    core.addPath(toolPath);
     await cfg.createCluster();
   } catch (error) {
     core.setFailed(error.message);

